app/src/lib/repository-matching.ts

```diff
@@ -92,3 +95,47 @@ export function matchExistingRepository(
     }) || null
   )
 }
+
+/**
+ * Check whether or not a GitHub repository matches a given remote.
+ *
+ * The remote's URL might match the cloneURL field of the repository (with a
+ * .git suffix) or it might match the htmlURL field (without a .git suffix)
+ *
+ * @param gitHubRepository the GitHub repository details found from the GitHub API
+ * @param remote the remote details found in the Git repository
+ */
+export function repositoryMatchesRemote(
+  gitHubRepository: GitHubRepository,
+  remote: IRemote
+): boolean {
+  const htmlUrl = gitHubRepository.htmlURL || ''
```

Comment:
Granted, this is an academic discussion but Could we opt to not perform htmlURL checks if it's null rather than coalescing to an empty string?